dask_jobqueue/sge.py

```diff
@@ -82,3 +93,31 @@ def __init__(self, queue=None, project=None, resource_spec=None, walltime=None,
         self.job_header = header_template % config
 
         logger.debug("Job script: \n %s" % self.job_script())
+
+
+def parse_memory(memory: str):
```

Comment:
I think the input to this fonction should be worker_memory, that we optain using distributed.utils.parse_bytes. Memory kwarg can be a number with no explicit unit if I'm not mistaken. You should probably use format_memory from the same package, and base your conversation from its output.

Reply:
Got it!

Reply:
@guillaumeeb I did a search of dask/distributed for `format_memory`, but could not find it. Here's the URL to the search: https://github.com/dask/distributed/search?utf8=✓&q=format_memory&type=

Can you point to where the function `format_memory` is located?

Reply:
https://github.com/dask/distributed/blob/964a917c45e8c11b16086cf05949d3529cc0986a/distributed/utils.py#L1129-L1168

Reply:
Thanks @mrocklin! Just to confirm, then, @guillaumeeb, you were referring to `parse_bytes` when you wrote `format_memory`? Having found `parse_bytes`, I think I know how to move forward from here.

Reply:
You've got `format_bytes` just above, sorry the function name I used was wrong: https://github.com/dask/distributed/blob/964a917c45e8c11b16086cf05949d3529cc0986a/distributed/utils.py#L1081-L1107.

The only thing is that format_bytes uses "real" Giga/Mega/Kilobytes and note Gibi/Mebi/Kibibytes (see https://en.wikipedia.org/wiki/Gibibyte). For example, PBS uses the GB term, but its meaning is GiB. See https://github.com/dask/dask-jobqueue/blob/master/dask_jobqueue/pbs.py#L106-L128.  But I actually don't really like this part of the code, and since GiB>GB, I think it's fine to use Dask's GB as GiB for PBS/SGE, without doing a real conversion.